rbi/annotations/lhm-shopify.rbi

```diff
@@ -0,0 +1,198 @@
+end
+
+class Lhm::Printer::Base
+  sig { void }
+  def initialize; end
+end
+
+class Lhm::Printer::Percentage < Lhm::Printer::Base
+  sig { void }
+  def end; end
+
+  sig { params(lowest: Numeric, highest: T.nilable(Numeric)).void }
+  def notify(lowest, highest); end
+end
+
+class Lhm::Printer::Dot < Lhm::Printer::Base
+  sig { void }
+  def end; end
+
+  sig { params(_arg0: T.untyped).void }
+  def notify(*_arg0); end
+end
+
+class Lhm::Migrator
+  include Lhm::Command
+
+  sig { returns(String) }
+  attr_reader :name
+
+  sig { returns(T::Array[String]) }
+  attr_reader :statements
+
+  sig { returns(T.nilable(Lhm::Connection)) }
+  attr_reader :connection
+
+  sig { returns(T::Array[String]) }
+  attr_reader :conditions
+
+  sig { returns(T::Hash[T.untyped, T.untyped]) }
+  attr_reader :renames
+
+  sig { returns(Lhm::Table) }
+  attr_reader :origin
+
+  sig { params(table: Lhm::Table, connection: T.nilable(Lhm::Connection)).void }
+  def initialize(table, connection = nil); end
+
+  sig { params(name: T.any(String, Symbol), definition: T.any(String, Symbol)).void }
+  def add_column(name, definition); end
+
+  sig do
+    params(
+      columns: T.any(String, Symbol, T::Array[T.any(String, Symbol)]),
+      index_name: T.nilable(String, Symbol)
+    ).void
+  end
+  def add_index(columns, index_name = nil); end
+
+  sig do
+    params(
+      columns: T.any(String, Symbol, T::Array[T.any(String, Symbol)]),
+      index_name: T.nilable(String, Symbol)
+    ).void
+  end
+  def add_unique_index(columns, index_name = nil); end
+
+  sig { params(name: T.any(String, Symbol), definition: T.any(String, Symbol)).void }
+  def change_column(name, definition); end
+
+  sig { params(statement: String).void }
+  def ddl(statement); end
+
+  sig { params(sql: String).returns(String) }
+  def filter(sql); end
+
+  sig { params(name: T.any(String, Symbol)).void }
+  def remove_column(name); end
+
+  sig do
+    params(
+      columns: T.any(String, Symbol, T::Array[T.any(String, Symbol)]),
+      index_name: T.nilable(String, Symbol)
+    ).void
+  end
+  def remove_index(columns, index_name = nil); end
+
+  sig { params(old: T.any(String, Symbol), nu: T.any(String, Symbol)).void }
+  def rename_column(old, nu); end
+end
+
+module Lhm::Throttler
+  sig { params(type: T.any(Lhm::Command, Symbol, String, Class), options: T::Hash[T.untyped, T.untyped]).void }
+  def setup_throttler(type, options = {}); end
+
+  sig { returns(Lhm::Throttler) }
+  def throttler; end
+end
+
+class Lhm::Throttler::Time
+  include Lhm::Command
+
+  sig { returns(Integer) }
+  attr_reader :stride
+
+  sig { returns(Numeric) }
+  attr_reader :timeout_seconds
+
+  sig { params(options: T::Hash[T.untyped, T.untyped]).void }
+  def initialize(options = {}); end
+
+  sig { void }
+  def execute; end
+
+  sig { params(arg0: Integer).returns(Integer) }
+  def stride=(arg0); end
+
+  sig { params(arg0: Numeric).returns(Numeric) }
+  def timeout_seconds=(arg0); end
```

Comment:
I think you can change the `attr_reader` into `attr_accessor` and get rid of this: https://github.com/Shopify/lhm/blob/master/lib/lhm/throttler/time.rb#L9-L10